### PR TITLE
Precise mode for parsing numbers

### PR DESCRIPTION
**Pull request by grisumbras**
This is technically a fix to #599, but we want a better implementation than `strtod` (because it is very slow). Closer to the release I'll take some files from Charconv.

### CODE REVIEW

File: include/fast_float/parse_number.h [hunk 1]
```diff
@@ -0,0 +1,230 @@
+#ifndef FASTFLOAT_PARSE_NUMBER_H
+#define FASTFLOAT_PARSE_NUMBER_H
+
```
Comment (vinniefalco) [inline include/fast_float/parse_number.h:3]: You need copyrights and please "hpp" extension not "h"

File: include/fast_float/parse_number.h [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+#include "ascii_number.h"
+#include "decimal_to_binary.h"
+#include "digit_comparison.h"
```
Comment (vinniefalco) [inline include/fast_float/parse_number.h:6]: These need to be standard full boost paths, e.g. <boost/json/ umm... wait a minute you have this in <boost/fastfloat/> ? That's not allowed

File: include/fast_float/parse_number.h [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+#include "float_common.h"
+
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <system_error>
+
+namespace fast_float {
```
Comment (vinniefalco) [inline include/fast_float/parse_number.h:14]: Where's the boost namespace?

File: include/fast_float/parse_number.h (hunk 2 above)
Comment (grisumbras) [reply include/fast_float/parse_number.h]: Turns out fast_float isn't permissive enough, so I had to remove it anyway.

**Pull request merged**